### PR TITLE
Update azure-sdk-for-go to v21.0.0

### PR DESCRIPTION
**Pull request by mboersma**
**What this PR does / why we need it**: 
Not required, just trying to keep things fresh and following on to #3843.
See https://github.com/Azure/azure-sdk-for-go/releases/tag/v21.0.0

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
Update azure-sdk-for-go to v20.2.0
```

### CODE REVIEW

**Comment by tariq1890** [comment]
FYI @mboersma , this version does not include the latest go-autorest. I raised an issue for this. Looks like we'll have to wait for another release . https://github.com/Azure/azure-sdk-for-go/issues/2776

**Comment by mboersma** [comment]
@tariq1890 the v19.0.0 version of azure-sdk-for-go we currently have vendored in also doesn't specify the latest go-autorest. This version is only a patch release off from our ideal v10.15.3 and vendors in cleanly and passes tests, so I'm not sure why we're blocked waiting for another release.

**Comment by tariq1890** [comment]
Agreed! We are certainly not blocked. I had assumed that the intent of this PR was to bring the Go SDK and autorest in sync.

**Comment by mboersma** [comment]
Understood now. Well, not in sync, but closer at least. We can wait for the next v21.0.0 release before we vendor since it sounds like it's coming soon and will sync everything up.

**Comment by mboersma** [comment]
@tariq1890 I bumped this to v21.0.0 which was just released. Looks like it does use go-autorest [10.15.4](https://github.com/Azure/azure-sdk-for-go/blob/v21.0.0/Gopkg.toml#L25).

File: pkg/armhelpers/azureclient.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 	"github.com/Azure/azure-sdk-for-go/services/authorization/mgmt/2015-07-01/authorization"
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-04-01/compute"
 	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
-	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-05-01/network"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-08-01/network"
```
Comment (tariq1890) [inline pkg/armhelpers/azureclient.go]: Love this change. I was looking at the release notes and 2018-05-01/network is a deprecated api call :).

**Pull request merged**